### PR TITLE
[DB source errors] : Handle common transient errors

### PR DESCRIPTION
**Pull request by akashkulk**
- e.g. https://cloud.airbyte.com/workspaces/265a2998-b917-4b92-8d18-8aaa5008f7e8/connections/14780735-af83-43e2-a366-b83f45f2089e/job-history#10384358::0 : MySQL EOFException

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt [hunk 1]
```diff
@@ -49,6 +55,8 @@ object ConnectorExceptionUtil {
             RECOVERY_CONNECTION_ERROR_MESSAGE
         } else if (isUnknownColumnInFieldListException(e)) {
             e!!.message
+        } else if (isTransientError(e)) {
+            DATABASE_CONNECTION_ERROR
```
Comment (rodireich) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt:59]: Is transient always connection error going forward?

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt [hunk 2]
```diff
@@ -137,6 +145,16 @@ object ConnectorExceptionUtil {
         return e is ConnectionErrorException
     }
 
+    private fun isTransientEOFException(e: Throwable?): Boolean {
+        return (e is EOFException) &&
+            e.message!!.lowercase().contains("connection was unexpectedly lost")
```
Comment (rodireich) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt:150]: This is coming from error seen in logs I guess

**Review by rodireich — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt (hunk 1 above)
Comment (akashkulk) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionUtil.kt:59]: For now, yes. We can always tweak the setting going forward

**Pull request merged**